dags/analyzer_helper/tests/integration/test_discord_transform_raw_data.py

```diff
@@ -0,0 +1,300 @@
+                "content": "sample text",
+                "user_mentions": [],
+                "role_mentions": [],
+                "reactions": [
+                    "user1,user2,❤️",
+                ],
+                "replied_user": None,
+                "createdDate": self.period,
+                "messageId": "msg123",
+                "channelId": "channel456",
+                "channelName": "general-chat-name",
+                "threadId": "threadId123",
+                "threadName": "general-thread-name",
+                "isGeneratedByWebhook": False,
+            }
+        ]
+        expected_result = [
+            {
+                "actions": [],
+                "author_id": "user1",
+                "date": self.period,
+                "interactions": [
+                    {
+                        "name": "reaction",
+                        "type": "emitter",
+                        "users_engaged_id": ["user123"],
+                    }
+                ],
+                "metadata": {
+                    "bot_activity": True,
+                    "channel_id": "channel456",
+                    "thread_id": "threadId123",
+                },
+                "source_id": "msg123",
+            },
+            {
+                "actions": [],
+                "author_id": "user2",
+                "date": self.period,
+                "interactions": [
+                    {
+                        "name": "reaction",
+                        "type": "emitter",
+                        "users_engaged_id": ["user123"],
+                    }
+                ],
+                "metadata": {
+                    "bot_activity": False,
+                    "channel_id": "channel456",
+                    "thread_id": "threadId123",
+                },
+                "source_id": "msg123",
+            },
+            {
+                "actions": [{"name": "message", "type": "emitter"}],
+                "author_id": "user123",
+                "date": self.period,
+                "interactions": [
+                    {
+                        "name": "reaction",
+                        "type": "receiver",
+                        "users_engaged_id": ["user1", "user2"],
+                    },
+                ],
+                "metadata": {
+                    "bot_activity": False,
+                    "channel_id": "channel456",
+                    "thread_id": "threadId123",
+                },
+                "source_id": "msg123",
+            },
+        ]
+
+        result = self.transformer.transform(
+            raw_data=raw_data, platform_id=self.platform_id, period=self.period
+        )
+        self.assertEqual(result, expected_result)
+
+    def test_transform_data_empty(self):
+        raw_data = []
+
+        expected_result = []
+
+        result = self.transformer.transform(
+            raw_data=raw_data, platform_id=self.platform_id, period=self.period
```

Comment:
We don't need the `period` in transformation part. The transform would use the `createdDate` of messages to assign `date` to our processed data.